---
title: "Pl-KL Leftovers"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## These reads were already trimmed.

# DADA2
```{r}
library(dada2); packageVersion("dada2")
path <- "/Volumes/GoogleDrive/My\ Drive/Stromatolites_Eukaryome/Eukaryote_microbiome/raw-data/PL_KL_leftovers/paired" # CHANGE ME to the directory containing the fastq files after unzipping.
list.files(path)
```

```{r}
fnFs <- sort(list.files(path, pattern="_R1.fastq", full.names = TRUE))
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
plotQualityProfile(fnFs[1:3])
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq"))
names(filtFs) <- sample.names
out <- filterAndTrim(fnFs, filtFs, trimLeft=10, maxLen=600, minLen=100,
          maxN=0, maxEE=2,compress=FALSE, truncQ=2, rm.phix=TRUE, multithread=TRUE)
head(out)
errF <- learnErrors(filtFs, multithread=2)
plotErrors(errF, nominalQ=TRUE)
derepFs <- derepFastq(filtFs)
sam.names <- sapply(strsplit(basename(filtFs), "_"), `[`, 1)
names(derepFs) <- sam.names
ddFs <- dada(derepFs, err=NULL, selfConsist=TRUE, HOMOPOLYMER_GAP_PENALTY=-1, BAND_SIZE=32)
plotErrors(ddFs)
dadaFs <- dada(derepFs, err=ddFs[[1]]$err_out, pool=TRUE, multithread=2, HOMOPOLYMER_GAP_PENALTY=-1, BAND_SIZE=32)
dadaFs[[1]]
mergers <- makeSequenceTable(dadaFs)
head(mergers[[1]])
seqtab.all <- mergers
saveRDS(seqtab.all, "/Volumes/GoogleDrive/My\ Drive/Stromatolites_Eukaryome/Eukaryote_microbiome/analysis/DADA2/KL-PL-seqtab_forwards.rds") # CHANGE ME to where you want sequence 
```